sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowFunctionsExec.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.analysis.FunctionRegistry
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.util.StringUtils
+import org.apache.spark.sql.connector.catalog.FunctionCatalog
+import org.apache.spark.sql.execution.LeafExecNode
+
+/**
+ * Physical plan node for showing functions.
+ */
+case class ShowFunctionsExec(
+    output: Seq[Attribute],
+    catalog: FunctionCatalog,
+    namespace: Seq[String],
+    userScope: Boolean,
+    systemScope: Boolean,
+    pattern: Option[String]) extends V2CommandExec with LeafExecNode {
+
+  override protected def run(): Seq[InternalRow] = {
+    val rows = new ArrayBuffer[InternalRow]()
+    val systemFunctions = if (systemScope) {
+        FunctionRegistry.builtinOperators.keys.toSeq
+    } else Seq.empty
+    val userFunctions = if (userScope) {
+      catalog.listFunctions(namespace.toArray).map(_.name()).toSeq
+    } else Seq.empty
+    val allFunctions =
```

Comment:
I think we should include temp functions as well. Temp functions do not belong to any catalog and should always be printed.

Reply:
Let's me double check that v1 includes temp functions. The doc https://spark.apache.org/docs/3.3.0/sql-ref-syntax-aux-show-functions.html is poor. Not clear which scope what includes. Should the temp function be output in the USER scope or always independently from `function_kind`?

Reply:
can we check the v1 behavior?

Reply:
I have already checked, and added a test.